### ISSUE TITLE
Fatal error from pip prevented installation.

### ISSUE THREAD

**Issue by ZeroDot1** [malicialab/iocsearcher#3]
```py
[user@work ~]$ pipx install iocsearcher
Fatal error from pip prevented installation. Full pip output in file:
    /home/user/.local/pipx/logs/cmd_2023-06-18_23.57.31_pip_errors.log

pip failed to build package:
    pysha3

Some possibly relevant errors from pip install:
    error: subprocess-exited-with-error
    error: command '/usr/bin/gcc' failed with exit code 1

Error installing iocsearcher.
[user@work ~]$ 

```
```py
PIP STDOUT
----------
Collecting iocsearcher
  Downloading iocsearcher-1.0.0-py3-none-any.whl (28 kB)
Collecting beautifulsoup4 (from iocsearcher)
  Downloading beautifulsoup4-4.12.2-py3-none-any.whl (142 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 143.0/143.0 kB 1.5 MB/s eta 0:00:00
Collecting coinaddrvalidator (from iocsearcher)
  Downloading coinaddrvalidator-1.1.3-py2.py3-none-any.whl (27 kB)
Collecting langdetect (from iocsearcher)
  Downloading langdetect-1.0.9.tar.gz (981 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 981.5/981.5 kB 2.1 MB/s eta 0:00:00
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
Collecting lxml (from iocsearcher)
  Downloading lxml-4.9.2-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.manylinux_2_24_x86_64.whl (7.2 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 7.2/7.2 MB 5.2 MB/s eta 0:00:00
Collecting pdfminer.six (from iocsearcher)
  Downloading pdfminer.six-20221105-py3-none-any.whl (5.6 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 5.6/5.6 MB 10.5 MB/s eta 0:00:00
Collecting phonenumbers (from iocsearcher)
  Downloading phonenumbers-8.13.14-py2.py3-none-any.whl (2.6 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 2.6/2.6 MB 14.9 MB/s eta 0:00:00
Collecting python-magic (from iocsearcher)
  Downloading python_magic-0.4.27-py2.py3-none-any.whl (13 kB)
Collecting readabilipy (from iocsearcher)
  Downloading readabilipy-0.2.0-py3-none-any.whl (4.3 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 4.3/4.3 MB 6.4 MB/s eta 0:00:00
Collecting soupsieve>1.2 (from beautifulsoup4->iocsearcher)
  Downloading soupsieve-2.4.1-py3-none-any.whl (36 kB)
Collecting attrs>=17.4.0 (from coinaddrvalidator->iocsearcher)
  Downloading attrs-23.1.0-py3-none-any.whl (61 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 61.2/61.2 kB 15.9 MB/s eta 0:00:00
Collecting pysha3>=1.0.2 (from coinaddrvalidator->iocsearcher)
  Downloading pysha3-1.0.2.tar.gz (829 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 829.2/829.2 kB 20.1 MB/s eta 0:00:00
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
Collecting base58check>=1.0.1 (from coinaddrvalidator->iocsearcher)
  Downloading base58check-1.0.2-py2.py3-none-any.whl (6.0 kB)
Collecting zope.interface>=4.4.3 (from coinaddrvalidator->iocsearcher)
  Downloading zope.interface-6.0-cp311-cp311-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (249 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 249.0/249.0 kB 12.9 MB/s eta 0:00:00
Collecting blake256>=0.1.1 (from coinaddrvalidator->iocsearcher)
  Downloading blake256-0.1.1.tar.gz (9.4 kB)
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
Collecting cbor>=1.0.0 (from coinaddrvalidator->iocsearcher)
  Downloading cbor-1.0.0.tar.gz (20 kB)
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
Collecting bech32>=1.1.0 (from coinaddrvalidator->iocsearcher)
  Downloading bech32-1.2.0-py3-none-any.whl (4.6 kB)
Collecting groestlcoin-hash2>=1.1.1 (from coinaddrvalidator->iocsearcher)
  Downloading groestlcoin_hash2-1.1.1.tar.gz (4.8 kB)
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
Collecting six (from langdetect->iocsearcher)
  Downloading six-1.16.0-py2.py3-none-any.whl (11 kB)
Collecting charset-normalizer>=2.0.0 (from pdfminer.six->iocsearcher)
  Downloading charset_normalizer-3.1.0-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (197 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 197.3/197.3 kB 24.1 MB/s eta 0:00:00
Collecting cryptography>=36.0.0 (from pdfminer.six->iocsearcher)
  Downloading cryptography-41.0.1-cp37-abi3-manylinux_2_28_x86_64.whl (4.3 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 4.3/4.3 MB 18.3 MB/s eta 0:00:00
Collecting html5lib (from readabilipy->iocsearcher)
  Downloading html5lib-1.1-py2.py3-none-any.whl (112 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 112.2/112.2 kB 26.9 MB/s eta 0:00:00
Collecting regex (from readabilipy->iocsearcher)
  Using cached regex-2023.6.3-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (781 kB)
Collecting cffi>=1.12 (from cryptography>=36.0.0->pdfminer.six->iocsearcher)
  Downloading cffi-1.15.1-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (462 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 462.6/462.6 kB 26.4 MB/s eta 0:00:00
Requirement already satisfied: setuptools in ./.local/pipx/shared/lib/python3.11/site-packages (from zope.interface>=4.4.3->coinaddrvalidator->iocsearcher) (67.8.0)
Collecting webencodings (from html5lib->readabilipy->iocsearcher)
  Downloading webencodings-0.5.1-py2.py3-none-any.whl (11 kB)
Collecting pycparser (from cffi>=1.12->cryptography>=36.0.0->pdfminer.six->iocsearcher)
  Downloading pycparser-2.21-py2.py3-none-any.whl (118 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 118.7/118.7 kB 27.9 MB/s eta 0:00:00
Building wheels for collected packages: langdetect, blake256, cbor, groestlcoin-hash2, pysha3
  Building wheel for langdetect (setup.py): started
  Building wheel for langdetect (setup.py): finished with status 'done'
  Created wheel for langdetect: filename=langdetect-1.0.9-py3-none-any.whl size=993224 sha256=cac8320ee14302ee9468a13f5ecd09c2f173223587e3521e03bfcf68f9f7339b
  Stored in directory: /home/user/.cache/pip/wheels/0a/f2/b2/e5ca405801e05eb7c8ed5b3b4bcf1fcabcd6272c167640072e
  Building wheel for blake256 (setup.py): started
  Building wheel for blake256 (setup.py): finished with status 'done'
  Created wheel for blake256: filename=blake256-0.1.1-py3-none-any.whl size=9885 sha256=e7a80319fe0a0bd6dcf38267d526addb35ae58582540d1612e095cf0ecaca731
  Stored in directory: /home/user/.cache/pip/wheels/85/c0/d2/ac7c0e91ede18729bdba006c8fe635ab486192353f98cd7839
  Building wheel for cbor (setup.py): started
  Building wheel for cbor (setup.py): finished with status 'done'
  Created wheel for cbor: filename=cbor-1.0.0-cp311-cp311-linux_x86_64.whl size=54448 sha256=d0489215fd6c59b92f4136dbbe0a0b7c3cda1961c405bbafcc5ebff3109caebc
  Stored in directory: /home/user/.cache/pip/wheels/21/6b/45/0c34253b1af07d1d9dc524f6d44d74a6b191c43152e6aaf641
  Building wheel for groestlcoin-hash2 (setup.py): started
  Building wheel for groestlcoin-hash2 (setup.py): finished with status 'done'
  Created wheel for groestlcoin-hash2: filename=groestlcoin_hash2-1.1.1-py3-none-any.whl size=5406 sha256=648ed6e0fcdd04fc51b2135a0cde6e99fdff9364516ea99edcb5dd3789c95b42
  Stored in directory: /home/user/.cache/pip/wheels/ad/a7/e8/b69abe8fa31534dff5929adc5302de2282802eb5eda6b6ea25
  Building wheel for pysha3 (setup.py): started
  Building wheel for pysha3 (setup.py): finished with status 'error'
  Running setup.py clean for pysha3
Successfully built langdetect blake256 cbor groestlcoin-hash2
Failed to build pysha3

PIP STDERR
----------
  error: subprocess-exited-with-error
  
  × python setup.py bdist_wheel did not run successfully.
  │ exit code: 1
  ╰─> [18 lines of output]
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build/lib.linux-x86_64-cpython-311
      copying sha3.py -> build/lib.linux-x86_64-cpython-311
      running build_ext
      building '_pysha3' extension
      creating build/temp.linux-x86_64-cpython-311
      creating build/temp.linux-x86_64-cpython-311/Modules
      creating build/temp.linux-x86_64-cpython-311/Modules/_sha3
      gcc -DNDEBUG -g -fwrapv -O3 -Wall -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -ffat-lto-objects -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -fPIC -DPY_WITH_KECCAK=1 -I/home/user/.local/pipx/venvs/iocsearcher/include -I/usr/include/python3.11 -c Modules/_sha3/sha3module.c -o build/temp.linux-x86_64-cpython-311/Modules/_sha3/sha3module.o
      In Datei, eingebunden von Modules/_sha3/sha3module.c:20:
      Modules/_sha3/backport.inc:78:10: schwerwiegender Fehler: pystrhex.h: Datei oder Verzeichnis nicht gefunden
         78 | #include "pystrhex.h"
            |          ^~~~~~~~~~~~
      Kompilierung beendet.
      error: command '/usr/bin/gcc' failed with exit code 1
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for pysha3
ERROR: Could not build wheels for pysha3, which is required to install pyproject.toml-based projects

```

**Comment by malicialab** [comment]
It seems that the coinaddrvalidator library we use to validate the checksum of blockchain addresses depends on pysha3 which has been archived and itself uses some private functions that are no longer available in Python 3.11
The easiest option is roll Python back to 3.10 (3.10.6 seems to work fine)
The ideal solution would be replace the blockchain address validation for a better library (but did not find a better one for Python so  far) or our own code (but it may take a while)

Do you need to use the blockchain address extraction? A middle ground would be to make the blockchain address validation an extra package so that at least the rest of the functionality can be used in Python 3.11

**Comment by malicialab** [comment]
Coming back to this issue after a while. We have released a new iocsearcher version 2.0.0, in PyPI, which completely revamps the blockchain validation and removes the dependency on coinaddrvalidator, and thus on the problematic pysha3 package

The new iocsearcher version should install and work with Python 3.11.
If you are still interested in the tool and want to give it a try, let me know if the new version works for you
